src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -228,6 +235,17 @@ protected internal int GetInvocationCounter()
             return counter;
         }
 
+        /// <summary>
+        /// The implementation of System.Runtime.GetRandom.
+        /// Gets the next random number.
+        /// </summary>
+        /// <returns>The next random number.</returns>
+        protected internal BigInteger GetRandom()
+        {
+            nonceData = Cryptography.Helper.Murmur128(nonceData, ProtocolSettings.Network);
```

Comment:
I still think that merkle tree should be used too, in this way at least the primary will require a brute force to add a transaction, no like now

Reply:
> I still think that merkle tree should be used too, in this way at least the primary will require a brute force to add a transaction, no like now

this is a temperary solution, ngd is working on a bls solution.

Reply:
> I still think that merkle tree should be used too, in this way at least the primary will require a brute force to add a transaction, no like now

We're planning to add a fulling BLS solution. However, since we'd like to launch mainnet next month, not much time left. So at least we must ensure `nonce` in header and then we could apply BLS later, should need several months to make it complete.